File: app/src/main/java/net/tjado/usbgadget/LogFragment.java
package net.tjado.usbgadget;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.fragment.app.Fragment;

public class LogFragment extends Fragment {

    public LogFragment() {}

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_log, container, false);
        TextView logText = (TextView) v.findViewById(R.id.log);

        logText.setText(Log.getLog());
        Log.onLogChangeListener = () -> {
            getActivity().runOnUiThread(new Runnable() {

                @Override
                public void run() {
                    logText.setText(Log.getLog());
                }
            });

        };

        return v;
    }
}

File: app/src/main/java/net/tjado/usbgadget/LogFragment.kt
package net.tjado.usbgadget

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import net.tjado.usbgadget.Log.OnLogChangeListener
import net.tjado.usbgadget.databinding.FragmentLogBinding


class LogFragment : Fragment() {
    private var _binding: FragmentLogBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentLogBinding.inflate(inflater, container, false)
        val view = binding.root

        binding.log.text = Log.getLog()
        Log.onLogChangeListener = OnLogChangeListener {
            activity?.runOnUiThread { binding.log.text = Log.getLog() }
        }
        return view
    }
}